### PR TITLE
Fixing mail_id parsing + adding module.exports if possible

### PR DESCRIPTION
**Pull request by nedarb**
Support module.exports when present, otherwise set the window.Gmail instance.

Also add missing semicolons (jshint warnings)

### CODE REVIEW

**Comment by nedarb** [comment]
@KartikTalwar here's the updated PR

**Comment by josteink** [comment]
The best way to get a PR accepted is usually to make it constrained to only fixing *one* thing.

And in that regard, you can leave out making things Node-friendly, because that was fixed seperately in https://github.com/KartikTalwar/gmail.js/pull/323 which is out on npm now :)

File: src/gmail.js [hunk 1]
```diff
@@ -213,22 +213,25 @@ var Gmail_ = function(localJQuery) {
...
 
 
   api.check.is_google_apps_user =function() {
     var email = api.get.user_email();
     return email.indexOf('gmail.com', email.length - 'gmail.com'.length) == -1;
   };
 
+  api.check.is_in_popout = function() {
+    return api.tools.parse_url(window.location.href).view === 'btop';
+  };
```
Comment (josteink) [inline src/gmail.js]: This method should also be added to the documentation (and if possible also to the TypeScript type declarations).

File: src/gmail.js [hunk 2]
```diff
@@ -370,7 +373,8 @@ var Gmail_ = function(localJQuery) {
       }
     }
     else {
-      hash = api.tools.parse_url(window.location.href).th;
+      hash = api.tools.parse_url(window.location.href).th || 
+        window.location.hash.split("/").pop().replace(/#/, '').split('?')[0];
     }
```
Comment (josteink) [inline src/gmail.js]: Again, some comments or context on what's getting fixed here would be helpful.

**Comment by josteink** [comment]
@nedarb: I'm very much in support for fixing jshint/eslint errors. Appreciate the effort.

That said: I've created a separate issue for doing it 100% here: https://github.com/KartikTalwar/gmail.js/issues/327

Feel free to chime in with your opinion there :)

With the NodeJS part already merged separately, what remains here is then a few unspecified fixes, which is very hard to have an opinion on without further context.

If you provide that, I can consider merging this, but otherwise I think it's better to close it.

**Comment by josteink** [comment]
Both issues mentioned in PR subject has been fixed in other, merged PRs. Closing this one.

**Pull request closed without merging**